### DISCUSSION TITLE
Error: NEXT_REDIRECT  coming up when using redirect from 'next/navigation'

### DISCUSSION THREAD

**Discussion by Arpitvicky** [vercel/next.js#62822] [Help]
### Summary

I have a situation where I am using redirect method of 'next/navigation' to redirect the users to correct path programatically in server component (Next  14.1.1-canary.62 version). 
The redirection is happening well on the browsers but there always comes the below error in Next js server logs. The target is to get rid of those logs. Could you please help to fix it ?

### Additional information

Code is pretty simple 
```js
  if (isInValidUrl) { // this boolean is decided with a simple utility function
    redirect(otherPath);
  }
```

```js
Error: NEXT_REDIRECT


 digest: 'NEXT_REDIRECT;replace;/url/url/url;307;',
  mutableCookies: f {
    _parsed: Map(23) {
      'sp' => [Object],
      'persist%3AslideIn' => [Object],
      'persist%3Auser' => [Object],
      'reduxPersistIndex' => [Object],
      '__hstc' => [Object],
      'hubspotutk' => [Object],
      '__hssrc' => [Object],
      'persist%3AcookieDashboard' => [Object],
      'tracking-preferences' => [Object],
      'ajs_anonymous_id' => [Object],
      '_gcl_au' => [Object],
      '_ga' => [Object],
      '_uetvid' => [Object],
      'locale' => [Object],
      'usprivacy' => [Object],
      '_uetsid' => [Object],
```

**Comment by icyJoseph** [comment]
Hi,

How are you logging? The thing is that, `redirect` does a sleight of hand, by actually throwing an error, or rather an exception.

This error is then caught by an error boundary, which checks if its redirect kind of exception. If it is, then the redirect is done. Otherwise the error is re-thrown. 

My guess is that you have some kind of global error listener? So perhaps you have to ignore errors, which have a `digest` property, which starts with NEXT_DIRECT?

**Reply by Arpitvicky** [reply]
I am not doing anything extra for logging, it's Next JS redirect.js. Isn't this piece of code responsible for showing the repetitive logs or may be I am wrong in my understanding. Everything is working fine on browser side. No issues with redirection.
https://github.com/vercel/next.js/blob/8f5107de16bb09f4f3ee0a5b1edb387d75a75292/packages/next/src/client/components/redirect.ts#L18

Also I have a question, do you think , like these repetitive logs can be the reason for lots of memory consumption in a pod (k8s) where the application is deployed ?

I will check for any global error listener , may be ignore the errors with digest property

**Comment by Arpitvicky** [comment]
I checked for error listener in application, it's just the error.tsx setup which shows up when something gone wrong on the page but all good as I said on browser. This error is happening only on server side logs. Please suggest any ways to suppress this.
